api-operator/pkg/mgw/uservolumes.go

```diff
@@ -0,0 +1,143 @@
+package mgw
+
+import (
+	wso2v1alpha1 "github.com/wso2/k8s-api-operator/api-operator/pkg/apis/wso2/v1alpha1"
+	"github.com/wso2/k8s-api-operator/api-operator/pkg/k8s"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"strings"
+)
+
+// mgw-deployment-configs
+const (
+	mgwDeploymentConfigMapName = "mgw-deployment-configs"
+	mgwConfigMaps              = "mgwConfigMaps"
+	mgwSecrets                 = "mgwSecrets"
+)
+
+type DeploymentConfig struct {
+	Name          string `yaml:"name"`
+	MountLocation string `yaml:"mountLocation"`
+	SubPath       string `yaml:"subPath"`
+	Namespace     string `yaml:"namespace,omitempty"`
+}
+
+var logDeploy = log.Log.WithName("mgw.userDeploymentVolume")
+
+// UserDeploymentVolume returns the deploy volumes and volume mounts with user defined config maps and secrets
+func UserDeploymentVolume(client *client.Client, api *wso2v1alpha1.API) ([]corev1.Volume, []corev1.VolumeMount,
+	error) {
+	var deployVolume []corev1.Volume
+	var deployVolumeMount []corev1.VolumeMount
+	var mgwDeployVol *v1.Volume
+	var mgwDeployMount *v1.VolumeMount
+	mgwDeploymentConfMap := k8s.NewConfMap()
+	errGetDeploy := k8s.Get(client, types.NamespacedName{Name: mgwDeploymentConfigMapName, Namespace: api.Namespace},
+		mgwDeploymentConfMap)
+	if errGetDeploy != nil && errors.IsNotFound(errGetDeploy) {
+		logDeploy.Info("Get mgw deployment configs", "from namespace", wso2NameSpaceConst)
+		//retrieve mgw deployment configs from wso2-system namespace
+		err := k8s.Get(client, types.NamespacedName{Namespace: wso2NameSpaceConst, Name: mgwDeploymentConfigMapName},
+			mgwDeploymentConfMap)
+		if err != nil && !errors.IsNotFound(err) {
+			logDeploy.Error(err, "MGW Deployment configs not defined")
```

Comment:
The log should be "Error while reading user volumes config" intead of "Not found"